src/routes/home/components/DisciplineCard.tsx

```diff
@@ -1,91 +1,49 @@
-import { useFocusEffect } from '@react-navigation/native'
-import React, { ReactElement, useCallback, useEffect, useState } from 'react'
-import * as Progress from 'react-native-progress'
+import React from 'react'
 import styled from 'styled-components/native'
 
 import Button from '../../../components/Button'
 import ErrorMessage from '../../../components/ErrorMessage'
 import Loading from '../../../components/Loading'
-import { ContentSecondary } from '../../../components/text/Content'
+import { ContentSecondary, ContentSecondaryLight } from '../../../components/text/Content'
 import { Subheading } from '../../../components/text/Subheading'
 import { BUTTONS_THEME } from '../../../constants/data'
-import { Discipline, Document } from '../../../constants/endpoints'
+import { ForbiddenError } from '../../../constants/endpoints'
 import labels from '../../../constants/labels.json'
-import theme from '../../../constants/theme'
+import { DisciplineRequestData, isTypeLoadProtected } from '../../../hooks/helpers'
 import { useLoadDiscipline } from '../../../hooks/useLoadDiscipline'
-import { loadDocuments } from '../../../hooks/useLoadDocuments'
-import useReadNextExercise from '../../../hooks/useReadNextExercise'
-import useReadProgress from '../../../hooks/useReadProgress'
-import { childrenLabel } from '../../../services/helpers'
-import { reportError } from '../../../services/sentry'
+import AsyncStorage from '../../../services/AsyncStorage'
 import Card from './Card'
-
-const ProgressContainer = styled.View`
-  display: flex;
-  flex-direction: row;
-  align-items: center;
-  padding: ${props => props.theme.spacings.sm} 0 ${props => props.theme.spacings.xs};
-`
+import CustomDisciplineDetails from './CustomDisciplineDetails'
+import ProfessionDetails from './ProfessionDetails'
 
 const LoadingContainer = styled.View`
   padding-top: ${props => props.theme.spacings.xxl};
 `
 
-const NumberText = styled(Subheading)`
+const ErrorMessageForbidden = styled(ContentSecondaryLight)`
+  padding-top: ${props => props.theme.spacings.xl};
+`
+
+export const NumberText = styled(Subheading)`
   font-size: ${props => props.theme.fonts.headingFontSize};
   padding: ${props => props.theme.spacings.xs};
 `
 
-const UnitText = styled(ContentSecondary)`
+export const UnitText = styled(ContentSecondary)`
   font-size: ${props => props.theme.fonts.headingFontSize};
 `
 
-const ButtonContainer = styled.View`
+export const ButtonContainer = styled.View`
   margin: ${props => props.theme.spacings.xxs} auto;
 `
```

Comment:
Should not live here if used in other components imo.

Reply:
will keep as no better options